api/v1alpha1/akodeploymentconfig_webhook.go

```diff
@@ -230,19 +247,33 @@ func (r *AKODeploymentConfig) validateAviSecret(secret *corev1.Secret, secretRef
 
 // validateAviControllerVersion checks NSX Advanced Load Balancer controller version valid or not
 func (r *AKODeploymentConfig) validateAviControllerVersion() (string, *field.Error) {
-	controllerVersion := AVI_VERSION
+	controllerVersion := ""
 	if r.Spec.ControllerVersion != "" {
-		controllerVersion = r.Spec.ControllerVersion
-	}
-	var re = regexp.MustCompile(controllerVersionRegex)
-	if !re.MatchString(controllerVersion) {
-		return controllerVersion, field.Invalid(field.NewPath("spec", "ControllerVersion"),
-			r.Spec.ControllerVersion,
-			"invalid controller version format, example valid controller version: 20.1.3")
+		controllerVersion := r.Spec.ControllerVersion
+		var re = regexp.MustCompile(controllerVersionRegex)
+		if !re.MatchString(controllerVersion) {
+			return controllerVersion, field.Invalid(field.NewPath("spec", "ControllerVersion"),
+				r.Spec.ControllerVersion,
+				"invalid controller version format, example valid controller version: 20.1.3")
+		}
 	}
 	return controllerVersion, nil
 }
 
+// validateAviAccount checks if using inputs can connect to avi controller or not
```

Comment:
in the comment mention the fact that we are creating an AVI client with `username, password, cert, version` tuple

Reply:
Will update it in follow-up PR